src/modm/driver/display/ssd1306_i2c.hpp

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) 2014, 2016-2017, Sascha Schade
+ * Copyright (c) 2014-2016, 2018, Niklas Hauser
+ * Copyright (c) 2021, Thomas Sommer
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#ifndef MODM_SSD1306_I2C_HPP
+#define MODM_SSD1306_I2C_HPP
+
+#include <modm/architecture/interface/i2c_device.hpp>
+#include <modm/architecture/utils.hpp>
+#include <modm/processing/timer.hpp>
+#include <modm/ui/display/monochrome_graphic_display_vertical.hpp>
+
+#include "ssd1306_common.hpp"
+
+namespace modm
+{
+
+/// @ingroup modm_driver_ssd1306
+struct ssd1306_i2c
+{
+public:
+	enum class
+	Transfer : uint8_t
+	{
+		COMMAND_BURST = 0x00,
+		DATA_BURST = 0x40,
+		COMMAND = 0x80,
+		DATA = 0xC0
+	};
+
+	/// @cond
+	class Ssd1306_I2cWriteTransaction : public modm::I2cWriteTransaction
+	{
+	public:
+		Ssd1306_I2cWriteTransaction(uint8_t address);
+
+		bool
+		configureDisplayWrite(const uint8_t *buffer, std::size_t size);
+
+	protected:
+		virtual Writing
+		writing() override;
+
+		virtual void
+		detaching(modm::I2c::DetachCause cause) override;
+
+		inline bool
+		isWritable()
```

Comment:
```suggestion
		inline bool
		isWritable() const
```

Any member function that doesn't mutate object state should be declared as `const`.